package/pyproject.toml

```diff
@@ -17,7 +17,7 @@ requires = [
     # For unreleased versions of Python there is currently no known supported
     # NumPy version. In that case we just let it be a bare NumPy install
     "numpy<2.0; python_version>='3.12'",
-    "setuptools",
+    "setuptools>=68.0",
```

Comment:
> Ok we're blocked by NumPy funnily enough, they have a pin on a too old version of setuptools for 1.25.2 (it's a weird world).

That doesn't sound right, the first NumPy version to support Python `3.12.0rc1` is the beta release `1.26.0b1` I used to build SciPy wheels, which should drop `setuptools` for `meson`.

Two tricky things to consider that should help out:

1. `3.12` won't pickup the release candidate in `pyproject.toml` logic so you'll need: https://github.com/numpy/numpy/issues/23808#issuecomment-1678677094
2. You may also need `PIP_PRE` per: https://github.com/numpy/numpy/issues/23808#issuecomment-1677596759

Anyway, hope that helps, but NumPy shouldn't be holding you back, should just be a matter of the tweaks mentioned there.

Reply:
That's what I did for example: https://github.com/scipy/scipy/blob/maintenance/1.11.x/pyproject.toml#L50

(actually I'm not sure I should have hard pinned like that, but anyway it worked)

Reply:
And `PIP_PRE` is set here: https://github.com/scipy/scipy/blob/maintenance/1.11.x/.github/workflows/wheels.yml#L136

(using that extra index shouldn't be needed anymore though I don't think since the NumPy beta is on PyPI)

Reply:
Thanks @tylerjereddy that's super useful info!

I think here I was mostly looking to see "what bits of upstream are we waiting on for py3.12", rather than adding actual shims to get py3.12 to work.

What I might do is stick the py3.12rc under the nightly wheels cron checks (which turns off build isolation) and see if it works there. If so, we can revisit adding it to normal CI checks when numpy 1.26 comes out.